packages/wallets/src/hooks/useInputATMFormatter.ts

```diff
@@ -53,7 +53,9 @@ const useInputATMFormatter = (inputRef: React.RefObject<HTMLInputElement>, initi
             setCaret(newCaretPosition);
             setCaretNeedsRepositioning(true);
 
-            if (maxDigits && input.value.replace(separatorRegex, '').replace(/^0+/, '').length > maxDigits) return;
+            const inputDigitsCount = input.value.replace(separatorRegex, '').replace(/^0+/, '').length;
+            const changeDigitsCount = e.target?.value?.replace(separatorRegex, '').replace(/^0+/, '').length ?? 0;
+            if (maxDigits && changeDigitsCount >= inputDigitsCount && inputDigitsCount > maxDigits) return;
```

Comment:
hey @rostislav-deriv 
could we add a small comment related to what is happening here?

Reply:
very much agree with the above - comment or somehow name it / split it into multiple concidtions to make it readable

Reply:
Yea agreed, I'll add

Reply:
Done